wasmplugin/plugin.go

```diff
@@ -270,23 +281,23 @@ func (ctx *httpContext) OnHttpRequestBody(bodySize int, endOfStream bool) types.
 
 func (ctx *httpContext) OnHttpResponseHeaders(numHeaders int, endOfStream bool) types.Action {
 	defer logTime("OnHttpResponseHeaders", currentTime())
+	tx := ctx.tx
 
-	if ctx.interruptionHandled {
-		// Handling the interruption (see handleInterruption) generates a HttpResponse with the required status code.
+	if ctx.interruptionHandledPhase == PhaseRequestHeaders || ctx.interruptionHandledPhase == PhaseRequestBody {
```

Comment:
BTW, from what I understand, because this is the response headers callback, this new check is equivalent to `interruptionHandled` - if there were an interruption already, it could only have been one of those two phases, unless this callback is actually called multiple times for a single request...

So you may consider simplifying back to the old check, some of the other locations look a bit more complicated now, notably the `ActionPause` fallback logic that seems never executed could go away.